mesa_geo/geospace.py

```diff
@@ -48,6 +44,21 @@ def __init__(self, crs="epsg:3857"):
         self.idx = index.Index()
         self.idx.agents = {}
 
+    @property
+    def crs(self):
+        return self._crs
+
+    @property
+    def transformer(self):
```

Comment:
I didn't notice immediately that `self.transformer` is actually set in the new code inside the `self.crs = ...` during `__init__`. I think it would be helpful to add a docstring here that says `self.transformer` is always set together during the `self.crs` set or something. I can see this avoids the pitfall where people set their `self.crs` but forget to set the `self.transformer`.

Reply:
The `transformer` is mainly used to convert `GeoSpace` and `GeoAgent` into GeoJSON for frontend visualization, in `GeoAgent.__geo_interface__` and `MapModule.render()`. I think ideally it should be hidden from the users, probably as `self._transformer` and used in `__geo_interface__` or some `to_GeoJSON()` functions. We can handle this properly in a separate PR in the future.

But yeah docstrings would be helpful to avoid confusions. I've updated them.

Reply:
Yes, I think there is not actually a need for the transformer to be associated directly with the GeoSpace?

Reply:
Yup, but, if we don't have `model.space.transformer`, then each `GeoAgent` would need its own `transformer` attribute. Most of the time, `GeoAgent` of the same type would have the same `crs`, so there'll be many identical `geo_agent.transformer` objects.

Otherwise we could have a static `CRS` for a `GeoAgent` class, with a static `GeoAgent.Transformer` attribute. However I don't know how to update all `GeoAgent` of the same type if this static `CRS` is changed. So I didn't go down this path.

Reply:
What I meant is that the transformer doesn't have to be a class attribute but could just be an independent function, or am I missing something? (Honest question, since I am not that familiar with the code and out of practice with Python in general)

Reply:
Hmm I would prefer to have `transformer` as an attribute (either with `GeoSpace` or with `GeoAgent`) rather than an independent function, since it is tied with their `crs` (i.e., the `crs` that transforms from).

Whenever the `crs` gets changed, the `transformer` needs to be updated accordingly.

Reply:
But couldn't the crs be one of the function arguments? It probably depends on the overhead of creating the transformer. But you are probably the expert of the code atm, so whatever suits you best (and with the space being responsible for crs management having the transformer there makes sense

Reply:
Oh please I'm no expert - I just barely started my journey in contributing to Mesa-Geo which was created by you : )

> (and with the space being responsible for crs management having the transformer there makes sense

Precisely.

Anyways also as you mentioned in another comment, since there's no complaint from users at the moment, we could wait for some feedbacks first and then decide.